### PR TITLE
feat(content-manager): warn when publishing an entry with draft relations

### PR DESCRIPTION
**Pull request by jhoward1994**
### What does it do?

Shows a pop up when you try to publish an entry that has draft relations.

### Why is it needed?

V5 features and prevent accidental publishing with draft relations.

### How to test it?

Provide information about the environment and the path to verify the behaviour.

### Related issue(s)/PR(s)

DX-1336

### CODE REVIEW

File: packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx [hunk 1]
```diff
@@ -503,15 +506,113 @@ const PublishAction: DocumentActionComponent = ({
...
+          id: getTranslation('error.records.fetch-draft-relatons'),
+          defaultMessage: 'An error occurred while fetching draft relations on this document.',
+        }),
+      });
+    }
+  }, [isErrorDraftRelations, toggleNotification, formatMessage]);
+
+  React.useEffect(() => {
+    let totalDraftRelations = 0;
+    const localDraftRelations = new Set();
```
Comment (jhoward1994) [inline packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx:539]: @Marc-Roig 👋🏻 this PR is FE stuff but this is the part I'd like to get your opinion on as it relates to the relations connecting syntax.

It's used to count any draft relations that are only stored in the FE edit view form based on 'connect' relation arrays

Comment (Marc-Roig) [reply packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx:539]: >FE edit view form based on 'connect' relation arrays

should this take into account the FE local relations + the ones stored in database?

File: packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx [hunk 2]
```diff
@@ -503,15 +506,113 @@ const PublishAction: DocumentActionComponent = ({
...
+     */
+    const traverseAndExtract = (data: { [field: string]: any }) => {
+      Object.entries(data).forEach(([key, value]) => {
+        if (key === 'connect' && Array.isArray(value)) {
+          extractDraftRelations({ connect: value });
+        } else if (typeof value === 'object' && value !== null) {
+          traverseAndExtract(value);
+        }
+      });
+    };
```
Comment (Marc-Roig) [inline packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx:568]: not aware of what the data object has, but does it work with nested relations inside compos and dynamic zones?

File: packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx [hunk 3]
```diff
@@ -503,15 +506,113 @@ const PublishAction: DocumentActionComponent = ({
...
+        }
+
+        if (data) {
+          totalDraftRelations += data.data;
+          setCountOfDraftRelations(totalDraftRelations);
+        }
+      };
+
+      checkForDraftRelations();
+    }
```
Comment (Marc-Roig) [inline packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx:604]: when does this effect run? Is it every time the form values change? or when the user tries to publish an entry?

Even with this if (!isLoadingDraftRelations) check, if this runs on every keypress on the edit view form, we might be spamming the server with many requests that migth not be necessary

File: packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx (hunk 2 above)
Comment (jhoward1994) [reply packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx:568]: Yep I've just tested those cases. I'll look at the e2e app template and see if that can be covered

File: packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx [hunk 4]
```diff
@@ -503,15 +506,96 @@ const PublishAction: DocumentActionComponent = ({
...
+      traverseAndExtract(formValues);
+      setLocalCountOfDraftRelations(localDraftRelations.size);
+    }
+  }, [documentId, modified, formValues, setLocalCountOfDraftRelations]);
+
+  React.useEffect(() => {
+    if (documentId && !isLoadingDraftRelations) {
+      const fetchDraftRelationsCount = async () => {
+        const { data, error } = await countDraftRelations({
+          collectionType,
```
Comment (Marc-Roig) [inline packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx:580]: is it possible for this to create any kind of infinite loop?

`countDraftRelations` updating `isLoadingDraftRelations` and retriggering the `coundDraftRelations` again?

Comment (jhoward1994) [reply packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx]: Depending on the loading state of the query wasn't causing and infinite loops or
unecessary backend calls that I could see. I think you're right that we don't
need this dependency though. I've removed it

**Review by Marc-Roig — APPROVED** [review]
logic looks good , did not qa!

**Review by Bassel17 — APPROVED** [review]
LGTM! and QAed seems to be working

**Pull request merged**